### PR TITLE
fix: resolve CodeQL security alerts (path-injection, request-forgery)

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

Fixes two CodeQL security alerts in alpamon.

- Resolves `go/request-forgery` by having `validateWebSocketURL` return a sanitized URL reconstructed from trusted config sources, so `dialer.Dial()` never receives raw user input.
- Resolves `go/path-injection` by splitting `makeArchive` return values into `(archivePath, cleanupPath, error)`, where `cleanupPath` is always derived from `os.TempDir()`, breaking the taint chain to `os.Remove`.

## Changes

- `pkg/runner/pty.go`: `validateWebSocketURL` now returns a sanitized URL with scheme and host taken from `config.GlobalSettings.ServerURL` (trusted). `dialer.Dial()` uses the sanitized URL instead of the raw input.
- `pkg/runner/ftp_test.go`: Updated test call sites to handle the new `(string, error)` return signature.
- `pkg/executor/handlers/file/file.go`: Split `makeArchive` return into `(archivePath, cleanupPath, error)`. `cleanupPath` is always `os.TempDir()`-based or empty, ensuring `os.Remove` is never called with a user-controlled path.

### Universal
- [ ] Code follows project style guidelines
- [ ] Self-reviewed the code
- [ ] No hardcoded secrets or credentials

### Go
- [ ] Error handling is complete
- [ ] Tests include edge cases
- [ ] `go mod tidy` executed

## Testing

Manually verified on a dev environment (`alpacax.dev.alpacon.io`) using an Ubuntu VM (UTM):
- Websh: Terminal sessions work correctly over WebSocket connections
- WebFTP: File upload and download operations function as expected

### CODE REVIEW

File: pkg/executor/handlers/file/file.go [hunk 1]
```diff
@@ -328,23 +332,25 @@ func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, rec
 		cmd.Dir = dirPath
 	} else {
 		if recursive {
-			archiveName = path + ".zip"
+			archiveName = filepath.Join(os.TempDir(), uuid.New().String()+".zip")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/executor/handlers/file/file.go:335]: This is an intentional design decision. The display name was deliberately removed in commit c47a26d — the server-side does not rely on the multipart filename for directory uploads, as it uses the separately transmitted metadata. Keeping the UUID-based temp path simplifies the cleanup logic and avoids path-injection risks.

File: pkg/executor/handlers/file/file.go [hunk 2]
```diff
@@ -328,23 +332,25 @@ func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, rec
...
 			cmd.Dir = filepath.Dir(path)
 		} else {
 			archiveName = path
+			// cleanupPath stays "" — single file, no temp archive to clean up
 		}
 	}
 
 	if bulk || recursive {
 		err := cmd.Run()
 		if err != nil {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/executor/handlers/file/file.go:348]: Good catch. Fixed — added `os.Remove(cleanupPath)` before returning the error so partial temp files are cleaned up on zip failure.

File: pkg/executor/handlers/file/file.go [hunk 3]
```diff
@@ -308,14 +307,19 @@ func (h *FileHandler) parsePaths(homeDirectory string, pathList []string) ([]str
 	return paths, isBulk, isRecursive, nil
 }
 
-// makeArchive creates a zip archive from the specified paths
-func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, recursive bool, sysProcAttr *syscall.SysProcAttr) (string, error) {
+// makeArchive creates a zip archive from the specified paths.
+// It returns the archive file path, a cleanup path (non-empty only for temp archives), and any error.
+// cleanupPath is always derived from os.TempDir() and never from user input,
+// ensuring os.Remove(cleanupPath) is safe from path-injection.
+func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, recursive bool, sysProcAttr *syscall.SysProcAttr) (string, string, error) {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/executor/handlers/file/file.go:314]: The path construction logic here is straightforward: bulk/recursive paths use `os.TempDir() + UUID`, single files keep the original path. The contract is self-evident from the code and the doc comment above the function. Adding a unit test for this would require either mocking the external `zip` binary or factoring out trivial path-join logic into a separate function, both of which add complexity without meaningful coverage gain.

File: pkg/executor/handlers/file/file.go [hunk 4]
```diff
@@ -328,23 +332,25 @@ func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, rec
...
 		if recursive {
-			archiveName = path + ".zip"
+			archiveName = filepath.Join(os.TempDir(), uuid.New().String()+".zip")
+			cleanupPath = archiveName
 			cmd = exec.CommandContext(ctx, "zip", "-r", archiveName, filepath.Base(path))
 			cmd.SysProcAttr = sysProcAttr
 			cmd.Dir = filepath.Dir(path)
 		} else {
 			archiveName = path
+			// cleanupPath stays "" — single file, no temp archive to clean up
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/executor/handlers/file/file.go]: Fixed — removed spaces around the em dash to match the repository writing conventions.

File: pkg/runner/pty.go [hunk 5]
```diff
@@ -76,16 +76,16 @@ func NewPtyClient(data protocol.CommandData, apiSession *scheduler.Session, mana
...
 		return err
 	}
-	var err error
 	dialer := websocket.Dialer{
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: !config.GlobalSettings.SSLVerify,
 		},
 	}
-	pc.conn, _, err = dialer.Dial(pc.url, pc.requestHeader)
+	pc.conn, _, err = dialer.Dial(sanitizedURL, pc.requestHeader)
```
Comment (geunwoonoh) [inline pkg/runner/pty.go:89]: **[Medium]** `pc.url` is never updated to the sanitized URL. Both `initializePtySession` and `recovery` pass `sanitizedURL` to `Dial`, but `pc.url` still holds the original (unsanitized) value.

If `pc.url` is referenced elsewhere later (e.g., logging, future reconnection logic), it would still carry the pre-validation URL. Consider assigning `pc.url = sanitizedURL` after validation to keep the struct state consistent.

```go
sanitizedURL, err := validateWebSocketURL(pc.url)
if err != nil {
    return err
}
pc.url = sanitizedURL
```

**Review by geunwoonoh — COMMENTED** [review]
Two medium-severity suggestions below.

File: pkg/executor/handlers/file/file.go [hunk 6]
```diff
@@ -308,14 +307,19 @@ func (h *FileHandler) parsePaths(homeDirectory string, pathList []string) ([]str
...
+// ensuring os.Remove(cleanupPath) is safe from path-injection.
+func (h *FileHandler) makeArchive(ctx context.Context, paths []string, bulk, recursive bool, sysProcAttr *syscall.SysProcAttr) (string, string, error) {
 	var archiveName string
+	var cleanupPath string
 	var cmd *exec.Cmd
 	path := paths[0]
 
 	if bulk {
-		archiveName = filepath.Dir(path) + "/" + uuid.New().String() + ".zip"
+		archiveName = filepath.Join(os.TempDir(), uuid.New().String()+".zip")
```
Comment (geunwoonoh) [inline pkg/executor/handlers/file/file.go:321]: **[Medium]** Archive location changed from the source directory to `os.TempDir()` (`/tmp`). This is correct for path-injection prevention, but introduces a behavioral difference:

- **Before**: archive created alongside the source files (same partition)
- **After**: archive created in `/tmp` (potentially a different partition with limited space)

On servers where `/tmp` is a small tmpfs mount and the uploaded files are large, this could cause failures that did not occur before. Worth verifying that target environments have sufficient `/tmp` space, or consider documenting this change for operators.

File: pkg/runner/pty.go (hunk 5 above)
Comment (jisung-02) [reply pkg/runner/pty.go:89]: Good catch. Added pc.url = sanitizedURL after validation in initializePtySession to keep the struct state consistent.

File: pkg/executor/handlers/file/file.go (hunk 6 above)
Comment (jisung-02) [reply pkg/executor/handlers/file/file.go:321]: Agreed. This was an intentional change to os.TempDir() to prevent path injection. Archives are removed immediately after upload via defer os.Remove, so there shouldn't be accumulation issues, but it could affect large file uploads on environments where /tmp is a small tmpfs mount. If there's an appropriate place in the docs, I'll document this behavioral change.

**Review by geunwoonoh — APPROVED** [review]
